backend/infrahub/core/manager.py

```diff
@@ -550,17 +661,52 @@ async def get_one_by_hfid(
         cls,
         db: InfrahubDatabase,
         hfid: list[str],
-        kind: str,
-        fields: Optional[dict] = None,
-        at: Optional[Union[Timestamp, str]] = None,
-        branch: Optional[Union[Branch, str]] = None,
-        include_source: bool = False,
-        include_owner: bool = False,
-        prefetch_relationships: bool = False,
-        account=None,
-        branch_agnostic: bool = False,
-        raise_on_error: Literal[False] = False,
-    ) -> Optional[Any]: ...
+        kind: type[SchemaProtocol],
+        raise_on_error: Literal[False] = ...,
+        fields: dict | None = ...,
+        at: Timestamp | str | None = ...,
+        branch: Branch | str | None = ...,
+        include_source: bool = ...,
+        include_owner: bool = ...,
+        prefetch_relationships: bool = ...,
+        account=...,
+    ) -> SchemaProtocol | None: ...
+
+    @overload
+    @classmethod
+    async def get_one_by_hfid(
+        cls,
+        db: InfrahubDatabase,
+        hfid: list[str],
+        kind: type[SchemaProtocol],
+        raise_on_error: Literal[True],
+        fields: dict | None = ...,
+        at: Timestamp | str | None = ...,
+        branch: Branch | str | None = ...,
+        include_source: bool = ...,
+        include_owner: bool = ...,
+        prefetch_relationships: bool = ...,
+        account=...,
+        branch_agnostic: bool = ...,
+    ) -> SchemaProtocol: ...
+
+    @overload
+    @classmethod
+    async def get_one_by_hfid(
+        cls,
+        db: InfrahubDatabase,
+        hfid: list[str],
+        kind: type[SchemaProtocol],
+        raise_on_error: bool,
+        fields: dict | None = ...,
+        at: Timestamp | str | None = ...,
+        branch: Branch | str | None = ...,
+        include_source: bool = ...,
+        include_owner: bool = ...,
+        prefetch_relationships: bool = ...,
+        account=...,
+        branch_agnostic: bool = ...,
+    ) -> SchemaProtocol: ...
```

Comment:
Is this correct? If raise_on_error is just a bool which can be False. Shouldn't we return an Optional[SchemaProtocol] in that case? Just above this we have an overload that returns SchemaProtocol if raise_on_error is set to True.

Reply:
technically this one will never match because we have both `Literal[True]` and `Literal[False]` defined above but it's required to make mypy happy, more information here ([link](https://stackoverflow.com/questions/59359943/how-to-write-typing-overload-decorator-for-bool-arguments-by-value))